Phantomjs processes are not killed correctly on error

We noticed all resources on our server were used up by phantom phantomjs processes (how ironic ;)). When investigating, we found at that stryker doesn't always kill the phantomjs process. It doesn't kill it properly when we see the following error

> [2017-03-14 14:20:23.500] [WARN] IsolatedTestRunnerAdapter - Received unexpected message from test runner worker process: "result" while current task is dispose. Ignoring this message.

The error can happen multiple times per run. Most of the time it will go wrong in an exponential fashion. In the beginning it fails one, then it fails a little bit more often and in the end all tests will fail with this error all the time. '

I think this is a raise condition because only a few runs have this trouble, not all runs.

**Versions**:
"grunt-stryker": "^0.1.0",
"stryker": "^0.5.6",
"stryker-api": "^0.4.2",
"stryker-html-reporter": "^0.3.0", 
"stryker-jasmine": "^0.1.0",
"stryker-karma-runner": "^0.3.3",

Below you can find the list of installed packages with their version.
├── grunt-stryker@0.1.0
├─┬ UNMET PEER DEPENDENCY stryker@0.5.6
├── stryker-api@0.4.2
├─┬ stryker-html-reporter@0.3.0
├── stryker-jasmine@0.1.0
├─┬ stryker-karma-runner@0.3.3
npm ERR! peer dep missing: stryker@^0.3.1, required by grunt-stryker@0.1.0

As you can see we are indeed using stryker-karma-runner 0.3.3. I'll update it and see if the problem still occurs than.

Some more information after a debug session:

> [2017-03-20 12:44:42.990] [ERROR] IsolatedTestRunnerAdapter - TRUE timeouthandler
Mutation testing  [==============                          ] 34% (ETC 1102.5s)[527 killed] [56 survived] [0 no coverage] [1 timeout] [0 error][2017-03-20 12:44:43.610] [DEBUG] EventRecorderReporter - Writing event onMutantTested to file reports/mutation/events/00668-onMutantTested.json
[2017-03-20 12:44:44.016] [WARN] IsolatedTestRunnerAdapter - Received unexpected message from test runner worker process: "result" while current task is dispose. Ignoring this message.
[2017-03-20 12:44:44.017] [ERROR] IsolatedTestRunnerAdapter - Received DisposeDone
[2017-03-20 12:44:44.017] [ERROR] IsolatedTestRunnerAdapter - Sturen SIGKILL: undefined

--- a/packages/stryker/src/isolated-runner/IsolatedTestRunnerAdapter.ts
+++ b/packages/stryker/src/isolated-runner/IsolatedTestRunnerAdapter.ts
@@ -116,7 +116,7 @@ export default class TestRunnerChildProcessAdapter extends EventEmitter implemen
     this.currentTask = new DisposeTask(MAX_WAIT_FOR_DISPOSE);
     this.sendDisposeCommand();
     return this.currentTask.promise
-      .then(() => this.workerProcess.kill('SIGKILL'));
+      .then(() => this.workerProcess.kill());
   }
 
   private sendRunCommand(options: RunOptions) {

--- a/packages/stryker/test/unit/isolated-runner/IsolatedTestRunnerAdapterSpec.ts
+++ b/packages/stryker/test/unit/isolated-runner/IsolatedTestRunnerAdapterSpec.ts
@@ -124,15 +124,17 @@ describe('IsolatedTestRunnerAdapter', () => {
         return expect(fakeChildProcess.send).to.have.been.calledWith(serialize({ kind: 'dispose' }));
       });
 
-      describe('and child process responses to dispose', () => {
+      describe('and child process responds to dispose', () => {
         beforeEach(() => {
           const promise = sut.dispose();
           receiveMessage({ kind: 'disposeDone' });
           return promise;
         });
 
-        it('should kill the child process', () =>
-          expect(fakeChildProcess.kill).to.have.been.calledWith());
+        it('should kill the child process', () => {
+          expect(fakeChildProcess.kill).to.not.have.been.calledWith('SIGKILL');
+          expect(fakeChildProcess.kill).to.have.been.called;
+        });
       });
 
       describe('and a timeout occurred', () => {
@@ -144,8 +146,10 @@ describe('IsolatedTestRunnerAdapter', () => {
           return promise;
         });
 
-        it('should kill the child process', () =>
-          expect(fakeChildProcess.kill).to.have.been.calledWith('SIGKILL'));
+        it('should kill the child process', () => {
+          expect(fakeChildProcess.kill).to.not.have.been.calledWith('SIGKILL');
+          expect(fakeChildProcess.kill).to.have.been.called;
+        });
       });
     });
     it('should reject any exceptions', () => {

